salt/cloud/clouds/hetzner.py

```diff
@@ -340,6 +340,31 @@ def create(vm_):
         automount=vm_.get("automount", None),
     )
 
+    __utils__["cloud.fire_event"](
+        "event",
+        "requesting instance",
+        "salt/cloud/{}/requesting".format(name),
+        args=__utils__["cloud.filter_event"](
+            "requesting", vm_, ["name", "profile", "provider", "driver"]
+        ),
+        sock_dir=__opts__["sock_dir"],
+        transport=__opts__["transport"],
+    )
+
+    while True:
+        server = client.servers.get_by_id(response.server.id)
+
+        if server.status == "running":
+            log.info("Server {} is up running now.".format(server.name))
+            break
+        else:
+            log.info(
+                "Waiting for server {} to be running: {}".format(
+                    server.name, server.status
+                )
+            )
+            time.sleep(1)
```

Comment:
Its possible that a user could be caught in this while loop indefinitely. Can we had a reasonable timeout here and break if we hit that timeout.